File: backend/app/schemas/log.py
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field

# 创建日志的基础模型
class LogBase(BaseModel):
    entity_type: Optional[str] = Field(None, description="实体类型：machine, project, user, system等")
    entity_id: Optional[int] = Field(None, description="实体ID")
    category: str = Field(..., description="日志分类：system, operation, status, security, error")
    operation: str = Field(..., description="操作类型：create, update, delete, deploy, start, stop, login等")
    title: str = Field(..., description="日志标题")
    content: Optional[str] = Field(None, description="详细内容")
    status: Optional[str] = Field(None, description="状态：success, failed, warning, info")
    data: Optional[Dict[str, Any]] = Field(None, description="额外数据")
    user_ip: Optional[str] = Field(None, description="操作者IP地址")

# 创建日志请求模型
class LogCreate(LogBase):
    user_id: Optional[int] = Field(None, description="操作用户ID，为空表示系统操作")

# 日志过滤请求模型
class LogFilter(BaseModel):
    entity_type: Optional[str] = None
    entity_id: Optional[int] = None
    category: Optional[str] = None
    operation: Optional[str] = None
    status: Optional[str] = None
    user_id: Optional[int] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    limit: int = 100
    offset: int = 0

# 日志详情响应模型
class LogInDB(LogBase):
    id: int
    user_id: Optional[int] = None
    created_at: datetime
    
    class Config:
        orm_mode = True

# 日志列表响应模型
class Log(LogInDB):
    username: Optional[str] = None  # 添加用户名，方便前端展示 

File: frontend/src/types/log.ts
/**
 * 日志相关类型定义
 */

// 日志对象类型
export interface Log {
  id: number;
  entity_type: string | null;
  entity_id: number | null;
  category: string;
  operation: string;
  title: string;
  content: string | null;
  status: string | null;
  data: any | null;
  user_id: number | null;
  username: string | null;
  user_ip: string | null;
  created_at: string;
}

// 日志过滤参数
export interface LogFilter {
  entity_type?: string;
  entity_id?: number;
  category?: string;
  operation?: string;
  status?: string;
  user_id?: number;
  start_date?: string;
  end_date?: string;
  skip?: number;
  limit?: number;
}

// 日志状态颜色映射
export const LogStatusColors = {
  success: '#52c41a',
  failed: '#ff4d4f',
  warning: '#faad14',
  info: '#1890ff',
};

// 日志分类映射
export const LogCategoryMap = {
  system: '系统',
  operation: '操作',
  security: '安全',
  status: '状态',
  error: '错误',
};

// 常见操作类型映射
export const LogOperationMap: Record<string, string> = {
  create: '创建',
  update: '更新',
  delete: '删除',
  login: '登录',
  logout: '登出',
  deploy: '部署',
  start: '启动',
  stop: '停止',
  check: '检查',
  upload: '上传',
  download: '下载',
}; 